### DISCUSSION TITLE
Setting Charge Start/End Through Automation

### DISCUSSION THREAD

**Discussion by StraponXG** [Pho3niX90/solis_modbus#73] [Q&A]
Hi, firstly thank you so much for developing this for HA! What a great integration. Here's my install, which has been up and running with great success for a while now:

![image](https://github.com/user-attachments/assets/958cbae5-282a-445b-ae4b-c79f38f19075)

My energy supplier provides occasional free energy sessions and I'd like to be able to add these to my charge/discharge schedule automatically. I'm pretty comfortable with how to build the logic in HA, but I keep hitting a wall with how to get the hour/minute information from a variable in an automation into the charge/discharge start/end hour/minute fields within the solis_modbus integration. I can do it manually through the integration page, or if they're presented in a dashboard, but I can't figure out how to push time from a variable in automation to something like 'time.solis_s6_solis_time_charging_charge_end_slot_1'.

I've even tried to use solis_modbus.solis_write_holding_register, which I assumed would give me direct access to the modbus communication, but I get an unknown error. Here's an example that should set the first charge slot hour to '9':

![image](https://github.com/user-attachments/assets/554f2617-a24a-4eca-9a63-09b96d06dcf1)

If you have any ideas, they'd be very welcome!

**Comment by ctepelea** [comment]
I am also trying to use this to automatically track the discharge current with the energy use. Do you know what could be the issue? Have you figured out the syntax?

Executed: 15 February 2025 at 21:29:50
Error: 'NodeDictClass' object has no attribute 'data'
Result:
params:
  domain: solis_modbus
  service: solis_write_holding_register
  service_data:
    address: 43142
    value: 172
  target: {}
running_script: false

<img width="818" alt="Screenshot 2025-02-15 at 21 30 08" src="https://github.com/user-attachments/assets/cc7d122a-4b23-434a-8614-0e2e1fdc799a" />

**Reply by Ian-Johnston** [reply]
You might glean some info about changing start/end times from my post here:
https://github.com/Pho3niX90/solis_modbus/issues/137
Ian.
